From fb9329b1b5d2fdb6d34104efb4c80eef6d61ed45 Mon Sep 17 00:00:00 2001
From: Mason Chang <mason-cw.chang@mediatek.com>
Date: Wed, 30 Apr 2025 10:51:09 +0800
Subject: [PATCH] net: ethernet: mtk_eth_soc: change napi poll weight to 256

This parameter will strongly affect the peak throughput of the SW path.
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c | 15 +++++++++------
 drivers/net/ethernet/mediatek/mtk_eth_soc.h |  1 +
 2 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/mediatek/mtk_eth_soc.c b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
index 29d6c7d..6b7e4c8 100644
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -6436,19 +6436,22 @@ static int mtk_probe(struct platform_device *pdev)
 	 */
 	init_dummy_netdev(&eth->dummy_dev);
 
-	netif_napi_add(&eth->dummy_dev, &eth->tx_napi, mtk_napi_tx);
-	netif_napi_add(&eth->dummy_dev, &eth->rx_napi[0].napi, mtk_napi_rx);
+	netif_napi_add_weight(&eth->dummy_dev, &eth->tx_napi, mtk_napi_tx, MTK_NAPI_WEIGHT);
+	netif_napi_add_weight(&eth->dummy_dev, &eth->rx_napi[0].napi, mtk_napi_rx,
+			      MTK_NAPI_WEIGHT);
 
 	if (MTK_HAS_CAPS(eth->soc->caps, MTK_RSS)) {
 		for (i = 1; i < MTK_RX_RSS_NUM; i++)
-			netif_napi_add(&eth->dummy_dev, &eth->rx_napi[MTK_RSS_RING(i)].napi,
-				       mtk_napi_rx);
+			netif_napi_add_weight(&eth->dummy_dev,
+					      &eth->rx_napi[MTK_RSS_RING(i)].napi,
+					      mtk_napi_rx, MTK_NAPI_WEIGHT);
 	}
 
 	if (eth->hwlro) {
 		for (i = 0; i < MTK_HW_LRO_RING_NUM; i++) {
-			netif_napi_add(&eth->dummy_dev, &eth->rx_napi[MTK_HW_LRO_RING(i)].napi,
-				       mtk_napi_rx);
+			netif_napi_add_weight(&eth->dummy_dev,
+					      &eth->rx_napi[MTK_HW_LRO_RING(i)].napi,
+					      mtk_napi_rx, MTK_NAPI_WEIGHT);
 		}
 	}
 
diff --git a/drivers/net/ethernet/mediatek/mtk_eth_soc.h b/drivers/net/ethernet/mediatek/mtk_eth_soc.h
index fbfc4c0..19b2d0d 100644
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.h
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.h
@@ -37,6 +37,7 @@
 #define MTK_DMA_SIZE(x)		(SZ_##x)
 #define MTK_FQ_DMA_HEAD		32
 #define MTK_FQ_DMA_LENGTH	2048
+#define MTK_NAPI_WEIGHT		256
 #define MTK_RX_ETH_HLEN		(VLAN_ETH_HLEN + VLAN_HLEN + ETH_FCS_LEN)
 #define MTK_RX_HLEN		(NET_SKB_PAD + MTK_RX_ETH_HLEN + NET_IP_ALIGN)
 #define MTK_DMA_DUMMY_DESC	0xffffffff
-- 
2.45.2

